### PR TITLE
Add UDP socket connection functionality

### LINKED ISSUE #3: Add UDP Support
The library should be able to interact with a Ping1D over UDP as well as over serial.

### PR DESCRIPTION
**Pull request by patrickelectric**
Fix #3

### CODE REVIEW

File: generate/templates/device.py.in [hunk 1]
```diff
@@ -30,27 +48,58 @@ class PingDevice(object):
...
+            self.iodev.connect(self.server_address)
+            self.iodev.setblocking(0)
+
+        except Exception as exception:
+            raise Exception("Failed to open the given UDP port: {0}".format(exception))
+
+    ##
+    # @brief Read available data from the io device
+    def read_io(self):
+        if type(self.iodev).__name__ == 'Serial':
```
Comment (jaxxzer) [inline generate/templates/device.py.in]: if self.iodev == None?

File: generate/templates/device.py.in [hunk 2]
```diff
@@ -69,7 +118,10 @@ class PingDevice(object):
     #
     # @return Number of bytes written
     def write(self, data):
-        return self.iodev.write(data)
+        if type(self.iodev).__name__ == 'Serial':
+            return self.iodev.write(data)
+        else: #Socket
```
Comment (Williangalvani) [inline generate/templates/device.py.in]:
```suggestion
        else: # Socket
```

Comment (patrickelectric) [reply generate/templates/device.py.in]: Done

File: generate/templates/device.py.in (hunk 1 above)
Comment (patrickelectric) [reply generate/templates/device.py.in]: Done

File: generate/templates/device.py.in [hunk 3]
```diff
@@ -30,27 +48,58 @@ class PingDevice(object):
...
-        self.parser = pingmessage.PingParser()
+    ##
+    # @brief Do the connection via an UDP link
+    #
+    # @param host: UDP server address (IPV4) or name
+    # @param port: port used to connect with server
+    #
+    def connect_udp(self, host: str = None, port: int =12345):
+        if host is None:
+            host = '' # Use all interfaces
```
[the remark this answers is not in the document]
Comment (patrickelectric) [reply generate/templates/device.py.in]: Done

File: generate/templates/device.py.in [hunk 4]
```diff
@@ -30,27 +48,58 @@ class PingDevice(object):
...
 
-        ## A helper class to take care of decoding the input stream
-        self.parser = pingmessage.PingParser()
+    ##
+    # @brief Do the connection via an UDP link
+    #
+    # @param host: UDP server address (IPV4) or name
+    # @param port: port used to connect with server
+    #
+    def connect_udp(self, host: str = None, port: int =12345):
```
[the remark this answers is not in the document]
Comment (patrickelectric) [reply generate/templates/device.py.in]: Done

File: generate/templates/device.py.in [hunk 5]
```diff
@@ -30,27 +48,58 @@ class PingDevice(object):
 
             ## Serial object for device communication
             self.iodev = serial.Serial(device_name, baudrate)
+            time.sleep(1)
```
[the remark this answers is not in the document]
Comment (patrickelectric) [reply generate/templates/device.py.in]: Done

File: generate/templates/device.py.in [hunk 6]
```diff
@@ -20,7 +21,24 @@ class PingDevice(object):
...
+        self.iodev = None
+        self.server_address = None
+
+    ##
+    # @brief Do the connection via an serial link
+    #
+    # @param device_name: Serial device name. E.g: /dev/ttyUSB0 or COM5
+    # @param baudrate: Connection baudrate used in the serial communication
+    #
+    def connect_serial(self, device_name: str, baudrate: int =115200):
```
[the remark this answers is not in the document]
Comment (patrickelectric) [reply generate/templates/device.py.in]: Done

**Comment by patrickelectric** [comment]
Corrections done

**Pull request merged**